storage/client/src/main/java/com/azure/storage/blob/BlobClient.java

```diff
@@ -91,7 +95,81 @@ public AppendBlobClient asAppendBlobClient() {
      *      A {@link PageBlobClient} to this resource.
      */
     public PageBlobClient asPageBlobClient() {
-        return new PageBlobClient(this.blobAsyncClient.blobAsyncRawClient.azureBlobStorage);
+        return new PageBlobClient(blobAsyncClient.asPageBlobAsyncClient());
+    }
+
+    /**
+     * Initializes a {@link ContainerClient} object pointing to the containing this blob is in. This method does
+     * not create a container. It simply constructs the URL to the container and offers access to methods relevant to
+     * containers.
+     *
+     * @return
+     *     A {@link ContainerClient} object pointing to the container containing the blob
+     */
+    public ContainerClient getContainerClient() {
+        return new ContainerClient(blobAsyncClient.getContainerAsyncClient());
+    }
+
+    /**
+     * Gets the URL of the blob represented by this client.
+     * @return the URL.
+     */
+    public URL getBlobUrl() {
+        return blobAsyncClient.getBlobUrl();
+    }
+
+    /**
+     * Opens a blob input stream to download the blob.
+     * <p>
+     *
+     * @return An <code>InputStream</code> object that represents the stream to use for reading from the blob.
+     *
+     * @throws StorageException
+     *             If a storage service error occurred.
+     */
+    public final BlobInputStream openInputStream() {
```

Comment:
Where is openOutputStream?

Reply:
output streams depend on the blob type so they are on block/append/page blob clients